DEVELOPMENT.md

```diff
@@ -10,6 +10,104 @@ This document outlines the process for setting up the development environment fo
 
 ## Setting up the Environment
 
+Decide first if you'd like a full buildsystem (on a VM) or a containerized dev environment.
+
+### VM as buildsystem
+Uses a Ubuntu 24.04 as base to run `chef` to setup all dependencies.
+The initial compilation is CPU intense and `16vcpu` are recommended.
+This was tested on GCP: a balanced disk of 500 GB and a VM type that supports nested virtualization should be chosen, as of writing (May 2025)
+`n2-standard-16` works well. 
+
+> [!Warning]
+>  The first build takes several hours and at least 160 Gb of space
+
+Turn on nested virtualization and avoid the use of `spot` VMs for the first build to avoid the very long first
+build interrupting. If you create the VMs as templates from an image, you can later switch to more cost-effective `spot` instances.
+
+
+
+```yaml
+advancedMachineFeatures:
+  enableNestedVirtualization: true
+```
+
+1) Install chef and some dependencies
+ 
+```
+sudo apt update sudo apt install -y git coreutils mkcert libnss3-tools screen libvirt-daemon-system libvirt-clients qemu-kvm virt-manager
+curl -L https://chefdownload-community.chef.io/install.sh | sudo bash
+```
+Now, on this VM, clone pixie (or your fork of it)
+
+```
+git clone https://github.com/pixie-io/pixie.git
+cd pixie/tools/chef
+sudo chef-solo -c solo.rb -j node_workstation.json
+sudo usermod -aG libvirt $USER
+```
+
+Make permanent the env loading via your bashrc
+```sh
+echo "source /opt/px_dev/pxenv.inc " >> ~/.bashrc
+```
+
+Put the baselrc into your homedir:
+```sh
+cp .bazelrc ~/.
+```
+In order to very significantly speed up your work, you may opt for a local cache directory. This can be shared between users of the VM, if both are part of the same group.
+Create a cache dir under <directory-path> like /tmp/bazel
+```sh
+sudo groupadd bazelcache
+sudo usermod -aG bazelcache $USER
+sudo mkdir -p <directory-path>
+sudo chown :bazelcache <directory-path>
+sudo chmod 2775 <directory-path>
+```
+
+2) Create/Use a registry you control and login
+   
+```sh
+docker login ghcr.io/<myregistry>
+```
+
+3) Make Minikube run and deploy a vanilla pixie
+
+If you added your user to the libvirt group (`sudo usermod -aG libvirt $USER`), this will now work (if you did this interactively: you need to refresh your group membership, e.g. by logout/login)
+```sh
+make dev-env-start
+```
+Deploy vanilla pixie (remote cloud)
+```sh
+sudo bash -c "$(curl -fsSL https://getcosmic.ai/install.sh)"
+export PX_CLOUD_ADDR=getcosmic.ai
+px auth
+px deploy -p=1Gi
+```
+For reference and further information https://docs.px.dev/installing-pixie/install-guides/hosted-pixie/cosmic-cloud
```

Comment:
Can we link to the documentation at https://docs.px.dev/installing-pixie/install-schemes/cli/ instead? Since our docs website is in a [separate repository](https://github.com/pixie-io/docs.px.dev), I worry that this could be missed if/when we make changes to our documentation.

In my opinion the important part to capture is that it requires deploying the Vizier first before a custom build can be deployed. I was thinking something along these lines:

> Before a custom build of Pixie can be deployed, you must install Pixie to your Kubernetes cluster first. Follow the steps [here](https://docs.px.dev/installing-pixie/install-schemes/cli/) and ensure Pixie is fully working before proceeding.

Reply:
This should now be addressed and vendor neutral